### PR TITLE
Refs [TASK][38402] Handle events, Get and handle data BasicFlightsListScreen

### PR DESCRIPTION
**Pull request by vinhmone**
## Related Tickets
- [#Task 38402: Handle events](https://edu-redmine.sun-asterisk.vn/issues/38402)
- [#Task 38403: Get and handle data](https://edu-redmine.sun-asterisk.vn/issues/38403)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListPresenter.kt [hunk 1]
```diff
@@ -1,9 +1,69 @@
...
+    private val view: BasicFlightsListContract.View,
+    private val repository: FlightRepository
 ) : BasicFlightsListContract.Presenter {
 
+    override fun getFlights(basicFlight: BasicFlight) {
+        view.showLoading()
+        repository.getFlights(basicFlight, object : OnDataCallBack<List<Flight>> {
+            override fun onSuccess(data: List<Flight>) {
+                if (data.isEmpty()) dataUnavailable()
+                else getLocation(data, basicFlight)
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListPresenter.kt]: k viết được 1 dòng thì nhớ cho ngoặc nhé

File: app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListFragment.kt [hunk 2]
```diff
@@ -37,7 +36,14 @@ class BasicFlightsListFragment :
     }
 
     override fun initData() {
+        val repository = RepositoryUtils.getFlightRepository()
+        presenter = BasicFlightsListPresenter(this, repository)
 
+        val bundle = this.arguments
+        if (bundle != null) {
+            val basicFlight = bundle.getParcelable<BasicFlight>(HomeFragment.DATA_BASIC_FLIGHT)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListFragment.kt]:
```suggestion
        val basicFlight = bundle?.getParcelable<BasicFlight>(HomeFragment.DATA_BASIC_FLIGHT)
```

File: app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListFragment.kt [hunk 3]
```diff
@@ -23,7 +22,7 @@ class BasicFlightsListFragment :
     private val flightListAdapter by lazy {
         BasicFlightListAdapter(
             mutableListOf(),
-            this::itemFlightClick
+            this::itemFlightClick,
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListFragment.kt]: thêm dấu phẩy lm j thế

Comment (vinhmone) [reply app/src/main/java/com/sun/findflight/ui/basicflightslist/BasicFlightsListFragment.kt]: trước e có bật Trailing commas theo convention của công ty, xong giờ quen luôn ạ :D

**Review by thonc-2135 — APPROVED** [review]

**Pull request merged**